umake/frameworks/ide.py

```diff
@@ -68,69 +68,66 @@ def __init__(self):
-                         description=_("Pure Eclipse Luna (4.4)"),
+                         description=_("Eclipse Java"),
                          category=category, only_on_archs=['i386', 'amd64'],
-                         download_page=None,
+                         download_page='https://www.eclipse.org/downloads/',
+                         checksum_type=ChecksumType.sha512,
                          dir_to_decompress_in_tarball='eclipse',
                          desktop_filename='eclipse.desktop',
                          required_files_path=["eclipse"],
                          packages_requirements=['openjdk-7-jdk'])
 
-    def download_provider_page(self):
-        """First, we need to fetch the MD5, then kick off the proceedings.
+        self.bits = '' if platform.machine() == 'i686' else 'x86_64'
 
-        This could actually be done in parallel, in a future version.
-        """
-        logger.debug("Preparing to download MD5.")
+    def download_sha512(self, sha512_url):
+        logger.debug("Preparing to Download SHA1")
 
-        arch = platform.machine()
-        if arch == 'i686':
-            md5_url = self.DOWNLOAD_URL_PAT.format(arch='', suf='.md5')
-        elif arch == 'x86_64':
-            md5_url = self.DOWNLOAD_URL_PAT.format(arch='-x86_64', suf='.md5')
-        else:
-            logger.error("Unsupported architecture: {}".format(arch))
-            UI.return_main_screen(status_code=1)
-
-        @MainLoop.in_mainloop_thread
         def done(download_result):
-            res = download_result[md5_url]
+            res = download_result[sha512_url]
 
             if res.error:
                 logger.error(res.error)
                 UI.return_main_screen(status_code=1)
 
             # Should be ASCII anyway.
-            md5 = res.buffer.getvalue().decode('utf-8').split()[0]
-            logger.debug("Downloaded MD5 is {}".format(md5))
-
-            logger.debug("Preparing to download the main archive.")
-            if arch == 'i686':
-                download_url = self.DOWNLOAD_URL_PAT.format(arch='', suf='')
-            elif arch == 'x86_64':
-                download_url = self.DOWNLOAD_URL_PAT.format(arch='-x86_64',
-                                                            suf='')
-            self.download_requests.append(DownloadItem(download_url, Checksum(ChecksumType.md5, md5)))
-            self.start_download_and_install()
+            self.sha512 = res.buffer.getvalue().decode('utf-8').split()[0]
+        DownloadCenter(urls=[DownloadItem(sha512_url, None)], on_done=done, download=False)
+        # It has to wait for the download and decode of the sha512sum
+        while self.sha512 is None:
+            pass
+        logger.debug("Downloaded SHA512 is {}".format(self.sha512))
+
+    def parse_download_link(self, line, in_download):
+        """Parse Eclipse download links"""
+        url, self.sha512 = (None, None)
+        if "eclipse-java" in line and self.bits in line:
```

Comment:
Maybe give it a try with the pastebin I gave you to file up the result, but yeah, at least, we will ensure we download a linux version :)